src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,47 @@
 
 class Animal {
   // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
+
+  static died(animal) {
+    const diedAnimalIndex = this.alive.indexOf(animal);
+
+    this.alive.splice(diedAnimalIndex, 1);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
-}
+  constructor() {
+    super();
+    this.hidden = false;
+  };
+
+  hide() {
+    this.hidden = true;
+  }
+};
 
 class Carnivore extends Animal {
   // write your code here
+  bite(herbivore) {
+    if (!herbivore.hidden && herbivore.hasOwnProperty('hidden')) {
```

Comment:
Instead of `hasOwnProperty` you can check if object is instance of needed class by `instanceof`